pkg/availability/multisigcollector/multisigcollector.go

```diff
@@ -39,7 +28,7 @@ type ModuleParams = common.ModuleParams
 // Whenever an availability certificate is requested, it pulls a batch from the mempool module,
 // sends it to all replicas and collects params.F+1 signatures confirming that
 // other nodes have persistently stored the batch.
-func NewModule(mc *ModuleConfig, params *ModuleParams, logger logging.Logger) (modules.PassiveModule, error) {
+func NewModule(mc ModuleConfig, params *ModuleParams, logger logging.Logger) (modules.PassiveModule, error) {
```

Comment:
Why deal directly with the struct and not the pointer? Any reason in particular? I suppose one reason is that the function should not change the struct, so better for it to deal with a copy. The drawback is performance: although the structs are not incredibly huge each call with `moduleConfig` copies the struct. 

Feel free to resolve with whatever decision upon reading this comment.

Reply:
The particular reason was that it slightly simplified the code (there actually were places where explicit copies were necessary) and I think it might even have been the case that while in most places it was passed by reference, it was occasionally still passed by value. I wanted to unify this. And I think that it makes more sense and is safer to pass it by value, so its contents are not overwritten by mistake somewhere. Regarding performance, as you said, the structs are very small, but, more importantly, for all practical performance purposes, they are used as good as never, so performance is definitely not a concern.